gate-api/src/main/java/com/netflix/spinnaker/gate/api/extension/HttpResponse.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.api.extension;
+
+import com.netflix.spinnaker.kork.annotations.Alpha;
+import lombok.Data;
+
+/**
+ * An {@code HttpResponse} represents the response from a request that has been handled by an {@code
+ * ApiExtension}.
+ */
+@Alpha
+@Data(staticConstructor = "of")
+public class HttpResponse {
```

Comment:
Same comment as with `HttpRequest` ... definitely need support for header propagation if nothing else.